src/Skia/Avalonia.Skia/GeometryImpl.cs

```diff
@@ -103,6 +114,36 @@ public ITransformedGeometryImpl WithTransform(Matrix transform)
         {
             return new TransformedGeometryImpl(this, transform);
         }
+        
+        /// <inheritdoc />
+        public bool TryGetPointAtDistance(double distance, out Point point)
+        {
+            if (EffectivePath is null)
+            {
+                point = new Point();
+                return false;
+            }
+            
+            var res = new SKPathMeasure(EffectivePath).GetPosition((float)distance, out var skPoint);
```

Comment:
Creating a new `SKPathMeasure` for each call is quite slow since it has to copy the path, build segment list all the time. Might be quite slow for bigger paths.

Reply:
thanks for the heads up, i wasnt sure if it was fast enough for allocating. But the thing now is that when to make a new SKPathMeasure? im not sure where i'll watch when the path changes.

Reply:
Allocating is probably the smallest problem here but that constructor does a lot of work internally.
Generally this geometry is immutable so you might be able to cache this on demand.

Reply:
we probably should enforce geometry immutability in another PR haha.. and make the immutable brushes/pens structs

Reply:
anyways i'll initialize on ctor then